torch/csrc/jit/passes/prepack_folding.cpp

```diff
@@ -0,0 +1,67 @@
+#include <stack>
+
+#include <torch/csrc/jit/api/module.h>
+#include <torch/csrc/jit/passes/constant_pooling.h>
+#include <torch/csrc/jit/passes/constant_propagation.h>
+#include <torch/csrc/jit/passes/prepack_folding.h>
+
+namespace torch {
+namespace jit {
+
+namespace {
+bool nodeMatchesPackingOps(Node* n,
+    const std::unordered_set<std::string>& op_white_list) {
+  return (op_white_list.count(n->kind().toQualString()) != 0);
+}
+} // namespace
+
+// Must run this pass after constant folding.
+void FoldPrePackingOps(script::Module& m,
+    const std::unordered_set<std::string>& foldable_prepacking_ops) {
+  // Since this pass can be called by quantization or other passes as well,
+  // we need to make sure we generate a unique "packed_weight_\d+"
+  // Thus static uid.
+  static int64_t uid = 0;
+
+  auto method = m.get_method("forward");
+  auto graph = method.graph();
+  std::stack<Block*> blocks_to_visit;
+  std::unordered_set<Node*> nodes_to_delete;
+  blocks_to_visit.push(graph->block());
+  std::string attr_name_base("packed_weight_");
+  while (!blocks_to_visit.empty()) {
+    Block* b = blocks_to_visit.top();
+    blocks_to_visit.pop();
+    for (Node* n : b->nodes()) {
+      if (foldable_prepacking_ops.count(n->kind().toQualString()) != 0) {
```

Comment:
this can be `isPackOp(node)` where `isPackOp` function  is the argument of `FoldPrePackingOps`

Reply:
Will make that change.

Reply:
Made all requested changes.